packages/flutter/lib/src/widgets/_window_positioner.dart

```diff
@@ -0,0 +1,510 @@
+  /// window will be positioned relative to. If a corner anchor is set
+  /// (e.g. [WindowPositionerAnchor.topLeft] or [WindowPositionerAnchor.bottomRight]),
+  /// the anchor point will be at the specified corner;
+  /// otherwise, the derived anchor point will be centered on the specified
+  /// edge, or in the center of the anchor rectangle if no edge is specified.
+  ///
+  /// This is used in conjunction with [childAnchor] to determine the final position
+  /// of the child window.
+  final WindowPositionerAnchor parentAnchor;
+
+  /// Defines the anchor point for the child window.
+  ///
+  /// The specified anchor is used to derive an anchor point that will be positioned
+  /// relative to the [parentAnchor]. If a corner anchor is set (e.g. [WindowPositionerAnchor.topLeft] or
+  /// [WindowPositionerAnchor.bottomRight]), the anchor point will be at the specified corner;
+  /// otherwise, the derived anchor point will be centered on the specified
+  /// edge, or in the center of the anchor rectangle if no edge is specified.
+  ///
+  /// This is used in conjunction with [parentAnchor] to determine the final position
+  /// of the child window.
+  final WindowPositionerAnchor childAnchor;
+
+  /// Specify the window position offset relative to the position of the
+  /// anchor on the anchor rectangle and the anchor on the child.
+  ///
+  /// For example if the anchor of the anchor rectangle is at (x, y), the window
+  /// has a [childAnchor] of [WindowPositionerAnchor.topLeft], and the [offset]
+  /// is (ox, oy), the calculated window position will be (x + ox, y + oy).
+  /// The offset position of the window is the one used for constraint testing.
+  /// See [constraintAdjustment].
+  ///
+  /// An example use case is placing a popup menu on top of a user interface
+  /// element, while aligning the user interface element of the parent window
+  /// with some user interface element placed somewhere in the popup window.
+  final Offset offset;
+
+  /// Defines how Flutter will adjust the position of the window if the unadjusted
+  /// position would result in the window being partly constrained by the platform.
+  ///
+  /// Whether a window is considered "constrained" is left to the platform
+  /// to determine. For example, the window may be partly outside the
+  /// output's 'work area', thus necessitating the child window's
+  /// position be adjusted until it is entirely inside the work area.
+  ///
+  /// The adjustments can be combined, according to a defined precedence:
+  ///
+  /// 1. [WindowPositionerConstraintAdjustment.flipX] and [WindowPositionerConstraintAdjustment.flipY]
+  /// 2. [WindowPositionerConstraintAdjustment.slideX] and [WindowPositionerConstraintAdjustment.slideY]
+  /// 3. [WindowPositionerConstraintAdjustment.resizeX] and [WindowPositionerConstraintAdjustment.resizeY]
+  ///
+  /// The first adjustment that results in the child window being entirely inside the work area will be picked.
```

Comment:
This is also explained on the docs for WindowPositionerConstraintAdjustment, maybe deduplicate it with a macro?